pkg/api/frontend_logging.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/inconshreveable/log15"
+)
+
+var frontendLogger = log.New("frontend")
+
+type frontendSentryExceptionValue struct {
+	Value      string            `json:"value,omitempty"`
+	Type       string            `json:"type,omitempty"`
+	Stacktrace sentry.Stacktrace `json:"stacktrace,omitempty"`
+}
+
+type frontendSentryException struct {
+	Values []frontendSentryExceptionValue `json:"values,omitempty"`
+}
+
+type FrontendSentryEvent struct {
```

Comment:
This doesn't need exporting.
```suggestion
type frontendSentryEvent struct {
```